edward/stats/distributions.py

```diff
@@ -164,23 +188,18 @@ def entropy(self, mean=None, cov=1):
 
         Arguments
         ----------
-        mean: tf.Tensor, optional
+        mean: np.array or tf.Tensor, optional
             vector. Defaults to zero mean.
-        cov: tf.Tensor, optional
+        cov: np.array or tf.Tensor, optional
             vector or matrix. Defaults to identity.
-
-        Returns
-        -------
-        tf.Tensor
-            scalar
         """
-        if cov == 1:
+        if cov is 1:
             d = 1
             det_cov = 1.0
         else:
-            cov = tf.cast(cov, dtype=tf.float32)
+            cov = tf.cast(tf.squeeze(tf.convert_to_tensor(cov)), dtype=tf.float32)
             d = get_dims(cov)[0]
-            if len(cov.get_shape()) == 1: 
+            if len(cov.get_shape()) == 1:
                 det_cov = tf.reduce_prod(cov)
             else:
                 det_cov = tf.matrix_determinant(cov)
```

Comment:
eek. it's a shame cholesky is not well supported in tensorflow (yet). this will be numerically unstable for some case in the near future; we should beware.